mhkit/mooring/general.py

```diff
@@ -0,0 +1,54 @@
+import xarray as xr
+
+
+def laylength(ds, depth, tolerance=0.25):
+    """Calculate the laylength of a mooring line over time.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Dataset containing x,y,z nodes (ie Node1px, Node1py, Node1pz)
+    depth : float
+        Depth of seabed (m)
+    tolerance : float, optional
+        Tolerance to detect first lift point from seabed, by default 0.25 meters
+
+    Returns
+    -------
+    xr.Dataset
+        Array containing the laylength at each time step.
+
+    Raises
+    ------
+    ValueError
+        Checks for mininum number of nodes necessary to calculate laylength.
+    """
+    assert isinstance(ds, xr.Dataset), 'ds must be of type xr.Dataset'
+    assert isinstance(depth, (float,int)), 'depth must be of type float or int'
+    assert isinstance(tolerance, (float,int)), 'tolerance must be of type float or int'
```

Comment:
Going forward we are no longer using assert statements in functions (see #242). In your case these would no be

```
    if not isinstance(ds, xr.Dataset):
        raise TypeError('ds must be of type xr.Dataset')
    if not isinstance(depth, (float, int)):
        raise TypeError('depth must be of type float or int')
    if not isinstance(tolerance, (float, int)):
        raise TypeError('tolerance must be of type float or int')
```